simtools/corsika/corsika_output.py

```diff
@@ -0,0 +1,1029 @@
+            incoming event. It can be passed in radians or degrees.
+            If not given, no rotation is performed.
+        zenith_angle: astropy.Quantity
+            Zenith angle to rotate the observational plane and obtain it perpendicular to the
+            incoming event. It can be passed in radians or degrees.
+            If not given, no rotation is performed.
+
+        Raises
+        ------
+        IndexError:
+            If the index or indices passed though telescope_index are out of range.
+        """
+
+        hist_num = 0
+        for i_tel_info, photons_info in zip(self.telescope_positions, photons):
+
+            if azimuth_angle is None or zenith_angle is None:
+                photon_x, photon_y = photons_info["x"], photons_info["y"]
+            else:
+                photon_x, photon_y = rotate(
+                    photons_info["x"],
+                    photons_info["y"],
+                    azimuth_angle,
+                    zenith_angle,
+                )
+
+            if self.single_telescopes is False:
+                hist_num = 0
+                # Adding the position of the telescopes to the relative position of the photons
+                # such that we have a common coordinate system.
+                photon_x = -i_tel_info["x"] + photon_x
+                photon_y = -i_tel_info["y"] + photon_y
+
+            if (
+                i_tel_info in self.telescope_positions[self.telescope_indices]
+                or self.single_telescopes is False
+            ):
+                self.hist_position[hist_num].fill(
+                    (photon_x * u.cm).to(u.m),
+                    (photon_y * u.cm).to(u.m),
+                    np.abs(photons_info["wavelength"]) * u.nm,
+                )
+
+                self.hist_direction[hist_num].fill(photons_info["cx"], photons_info["cy"])
+                self.hist_time_altitude[hist_num].fill(
+                    photons_info["time"] * u.ns, (photons_info["zem"] * u.cm).to(u.km)
+                )
+                hist_num += 1
+
+    def set_histograms(self, telescope_indices=None, single_telescopes=False):
+        """
+        Extract the information of the Cherenkov photons from a CORSIKA output IACT file, create
+         and fill the histograms
+
+        Parameters
+        ----------
+        telescope_indices: int or list of int
+            The indices of the specific telescopes to be inspected. If not specified, all telescopes
+            are treated together in one histogram.
+        single_telescopes: bool
+            if False, the histograms are supposed to be filled for all telescopes.
+            if True, one histogram is set for each telescope sepparately.
+
+        Returns
+        -------
+        list: list of boost_histogram.Histogram instances.
+
+        """
+        if telescope_indices is None:
+            telescope_indices = np.arange(self.num_telescopes).tolist()
```

Comment:
As I wrote in mattermost, it is better to have it here, since here is the first place where the user writes in which telescopes he/she is interested. it also allows one to change by setting it again here